azkaban-common/src/main/java/azkaban/utils/SystemMemoryInfo.java

```diff
@@ -0,0 +1,165 @@
+package azkaban.utils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.log4j.Logger;
+
+/**
+ * @author wkang
+ * This class is used to maintain system memory information. Processes utilizing
+ * large amount of memory should consult this class to see if the system has enough
+ * memory to proceed the operation.
+ * 
+ * Memory information is obtained from /proc/meminfo, so only Unix/Linux like system
+ * will support this class.
+ * 
+ * All the memory size used in this function is in KB
+ */
+public class SystemMemoryInfo {
+  private static final Logger logger = Logger.getLogger(SystemMemoryInfo.class);
+
+  private static String MEMINFO_FILE = "/proc/meminfo"; 
+  private static boolean memCheckEnabled;
+  private static boolean memInfoExists;
+  private static long freeMemAmount = 0;
+  private static long freeMemDecrAmt = 0;
+  private static final long LOW_MEM_THRESHOLD = 3L*1024L*1024L; //3 GB
+
+  private static ScheduledExecutorService scheduledExecutorService;
+
+  public static void init(boolean memChkEnabled, long memDecrAmt) {
+    File f = new File(MEMINFO_FILE);
+    memInfoExists = f.exists() && !f.isDirectory();
+    memCheckEnabled = memChkEnabled && memInfoExists;
+    if (memCheckEnabled) {
+      //initial reading of the mem info
+      readMemoryInfoFile();
+      freeMemDecrAmt = memDecrAmt;
+
+      //schedule a thread to read it
+      logger.info("Scheduled thread to read /proc/meminfo every 30 seconds");
+      scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+      scheduledExecutorService.scheduleAtFixedRate(new MemoryInfoReader(), 0, 30, TimeUnit.SECONDS);
+    }
+  }
+
+  public synchronized static boolean requestMemory(long xms, long xmx) {
```

Comment:
Given this is a static method, users/developers may not realize that an init needs to be called first.
In this case do you want to help the users/developers out by doing init for them, print a warning, or something of that sort?